isolate_example/lib/page_one.dart

```diff
@@ -12,20 +12,117 @@
 import 'package:flutter/material.dart';
 
-class PerformancePage extends StatelessWidget {
+class PerformancePage extends StatefulWidget {
+  @override
+  _PerformancePageState createState() => _PerformancePageState();
+}
+
+class _PerformancePageState extends State<PerformancePage> {
+  Future<void> computeFuture = Future.value();
+
   @override
   Widget build(BuildContext context) {
     return Center(
       child: Column(
         mainAxisSize: MainAxisSize.min,
-        children: <Widget>[
+        children: [
           SmoothAnimationWidget(),
+          Container(
+            alignment: Alignment.bottomCenter,
+            padding: EdgeInsets.only(top: 150),
+            child: Column(
+              children: [
+                FutureBuilder<void>(
+                  future: computeFuture,
+                  builder: (BuildContext context, AsyncSnapshot snapshot) {
+                    return RaisedButton(
+                      child: const Text('Compute on Main'),
+                      elevation: 8.0,
+                      onPressed: createMainIsolateCallBack(context, snapshot),
+                    );
+                  },
+                ),
+                FutureBuilder<void>(
+                  future: computeFuture,
+                  builder: (BuildContext context, AsyncSnapshot snapshot) {
+                    return RaisedButton(
+                      child: const Text('Compute on Secondary'),
+                      elevation: 8.0,
+                      onPressed:
+                          createSecondaryIsolateCallBack(context, snapshot),
+                    );
+                  },
+                ),
+              ],
+            ),
+          ),
         ],
       ),
     );
   }
+
+  VoidCallback createMainIsolateCallBack(
+      BuildContext context, AsyncSnapshot snapshot) {
+    if (snapshot.connectionState == ConnectionState.done) {
+      return () {
+        setState(() {
+          computeFuture = computeOnMainIsolate()
+            ..then((_) {
+              final snackBar = SnackBar(
+                content: Text('Main Isolate Done!'),
+              );
+              Scaffold.of(context).showSnackBar(snackBar);
+            });
+        });
+      };
+    } else {
+      return null;
+    }
+  }
+
+  VoidCallback createSecondaryIsolateCallBack(
+      BuildContext context, AsyncSnapshot snapshot) {
+    if (snapshot.connectionState == ConnectionState.done) {
+      return () {
+        setState(() {
+          computeFuture = computeOnSecondaryIsolate()
+            ..then((_) {
+              final snackBar = SnackBar(
+                content: Text('Secondary Isolate Done!'),
+              );
+              Scaffold.of(context).showSnackBar(snackBar);
+            });
+        });
+      };
+    } else {
+      return null;
+    }
+  }
+}
+
+Future<void> computeOnMainIsolate() async {
+  //The isolate will need a little time to disable the buttons before the performance hit
```

Comment:
nit: Comments should have a space after the slashes and end with a period (assuming they're complete sentences like this one).